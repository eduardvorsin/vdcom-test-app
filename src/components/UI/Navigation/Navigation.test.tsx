import React from 'react';

import { render, screen, waitFor } from '@testing-library/react';

import userEvent from '@testing-library/user-event';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import Navigation from './Navigation';
import wrapWithRedux from '../../../tests/helpers/wrapWithRedux';
import addMemoryRouter from '../../../tests/helpers/addMemoryRouter';
import server from '../../../tests/msw/server';

describe('Navigation tests', () => {
  it('renders correctly', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();

    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    render(
      wrapWithRedux(
        <RouterProvider router={mockRouter} />,
        {},
      ),
    );

    expect(screen.getByTestId<HTMLDivElement>('navigation')).toBeInTheDocument();
  });

  it('should display the logo if the width is greater than 900 pixels, and should not if less', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();

    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    const renderedElement = wrapWithRedux(
      <RouterProvider router={mockRouter} />,
      {},
    );

    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(1600);

    const { unmount } = render(renderedElement);

    expect(screen.getByAltText<HTMLImageElement>('logo')).toBeInTheDocument();

    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(768);

    unmount();
    render(renderedElement);

    expect(screen.queryByAltText<HTMLImageElement>('logo')).not.toBeInTheDocument();
  });

  it('should display the prev button if the width is less than 900 pixels, and should not if more', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();

    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    const renderedElement = wrapWithRedux(
      <RouterProvider router={mockRouter} />,
      {},
    );

    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(500);

    const { unmount } = render(renderedElement);

    expect(screen.getByRole<HTMLButtonElement>('button', { name: /close navigation/i })).toBeInTheDocument();

    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(1000);

    unmount();
    render(renderedElement);

    expect(screen.queryByRole<HTMLButtonElement>('button', { name: /close navigation/i })).not.toBeInTheDocument();
  });

  it('should remove the Navigation component from the body if the open prop is false', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();

    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={false}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    render(wrapWithRedux(
      <RouterProvider router={mockRouter} />,
      {},
    ));

    expect(screen.queryByTestId<HTMLDivElement>('navigation')).not.toBeInTheDocument();
  });

  it('should apply the width passed through the drawerWidth prop', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();

    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={432}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    render(wrapWithRedux(
      <RouterProvider router={mockRouter} />,
      {},
    ));

    expect(screen.getByTestId<HTMLDivElement>('navigation')).toHaveStyle('width:432px');
  });

  it('should call the mock for the onNavigationClose function when clicking the close navigation button', async () => {
    const user = userEvent.setup();
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();
    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={432}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    render(wrapWithRedux(
      <RouterProvider router={mockRouter} />,
      {},
    ));

    user.click(await screen.findByRole<HTMLButtonElement>('button', { name: /close navigation/i }));

    await waitFor(async () => {
      expect(mockFn).toHaveBeenCalledTimes(1);
    });
  });

  it('is a basic snapshot', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();
    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    render(
      wrapWithRedux(
        <RouterProvider router={mockRouter} />,
        {},
      ),
    );

    expect(screen.getByTestId<HTMLDivElement>('navigation')).toMatchSnapshot();
  });

  it('is a basic snapshot with a width of less than 900 pixels', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();
    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });
    jest.spyOn(window.screen, 'width', 'get').mockReturnValueOnce(500);

    render(
      wrapWithRedux(
        <RouterProvider router={mockRouter} />,
        {},
      ),
    );

    expect(screen.getByTestId<HTMLDivElement>('navigation')).toMatchSnapshot();
  });

  it('is a basic snapshot with a width of more than 900 pixels', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();
    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={true}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });
    jest.spyOn(window.screen, 'width', 'get').mockReturnValueOnce(1400);

    render(
      wrapWithRedux(
        <RouterProvider router={mockRouter} />,
        {},
      ),
    );

    expect(screen.getByTestId<HTMLDivElement>('navigation')).toMatchSnapshot();
  });

  it('is a snapshot with the open prop equal to false', () => {
    const mockFn = jest.fn<void, [e: React.MouseEvent<HTMLButtonElement>]>();
    const mockRouter = createMemoryRouter([{
      path: '/contacts',
      element: <Navigation
        drawerWidth={250}
        open={false}
        onNavigationClose={mockFn}
      />,
    }], {
      initialEntries: ['/contacts'],
    });

    render(
      wrapWithRedux(
        <RouterProvider router={mockRouter} />,
        {},
      ),
    );

    expect(screen.queryByTestId<HTMLDivElement>('navigation')).toMatchSnapshot();
  });
});

describe('Navigation integration tests', () => {
  beforeAll(() => {
    server.listen({ onUnhandledRequest: 'error' });
  });
  afterEach(() => {
    server.resetHandlers();
  });
  afterAll(() => {
    server.close();
  });

  it('should go to the login page when clicking on the logout button', async () => {
    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(1024);
    const user = userEvent.setup();
    const router = addMemoryRouter({
      initialEntries: ['/contacts'],
    });

    render(wrapWithRedux(router, {}));

    user.click(await screen.findByRole<HTMLAnchorElement>('link', { name: /Log out/i }));

    await waitFor(() => {
      expect(screen.queryByRole<HTMLAnchorElement>('link', { name: /Log out/i })).not.toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByRole<HTMLTitleElement>('heading', { name: /Welcome To CRM System Sign In To Your Account/i })).toBeInTheDocument();
    });
  });

  it('should go to the 404 page when clicking on calendar page link', async () => {
    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(1024);
    const user = userEvent.setup();
    const router = addMemoryRouter({
      initialEntries: ['/contacts'],
    });

    render(wrapWithRedux(router, {}));

    user.click(await screen.findByRole<HTMLAnchorElement>('link', { name: /calendar/i }));

    await waitFor(async () => {
      expect(screen.queryByRole<HTMLAnchorElement>('link', { name: /calendar/i })).not.toBeInTheDocument();
    });

    await waitFor(async () => {
      expect(screen.getByRole<HTMLTitleElement>('heading', { name: /404/i })).toBeInTheDocument();
    });
  });

  it('should go to the 404 page when clicking on project report page link', async () => {
    jest.spyOn(window.screen, 'width', 'get').mockReturnValue(1024);
    const user = userEvent.setup();
    const router = addMemoryRouter({
      initialEntries: ['/contacts'],
    });

    render(wrapWithRedux(router, {}));

    user.click(await screen.findByRole<HTMLAnchorElement>('link', { name: /project report/i }));

    await waitFor(() => {
      expect(screen.queryByRole<HTMLAnchorElement>('link', { name: /project report/i })).not.toBeInTheDocument();
    });
    await waitFor(async () => {
      expect(screen.getByRole<HTMLTitleElement>('heading', { name: /404/i })).toBeInTheDocument();
    });
  });
});
