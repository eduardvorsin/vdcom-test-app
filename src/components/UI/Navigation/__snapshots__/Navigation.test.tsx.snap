// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation tests is a basic snapshot 1`] = `
<div
  class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-1tvu1zq-MuiModal-root-MuiDrawer-root"
  data-testid="navigation"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft css-4t3x6l-MuiPaper-root-MuiDrawer-paper"
    tabindex="-1"
  >
    <div
      class="MuiBox-root css-1ndpvz5"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        close navigation
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <nav
      class="MuiBox-root css-10klw3m"
    >
      <ul
        class="MuiList-root MuiList-padding css-t39185-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            aria-current="page"
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root active"
            href="/contacts"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ContactsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 0H4v2h16V0zM4 24h16v-2H4v2zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-8 2.75c1.24 0 2.25 1.01 2.25 2.25s-1.01 2.25-2.25 2.25S9.75 10.24 9.75 9 10.76 6.75 12 6.75zM17 17H7v-1.5c0-1.67 3.33-2.5 5-2.5s5 .83 5 2.5V17z"
                />
              </svg>
            </span>
            Total Contacts
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
            href="/calendar"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DateRangeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                />
              </svg>
            </span>
            Calendar
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
            href="/project-report"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DataUsageIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 2.05v3.03c3.39.49 6 3.39 6 6.92 0 .9-.18 1.75-.48 2.54l2.6 1.53c.56-1.24.88-2.62.88-4.07 0-5.18-3.95-9.45-9-9.95zM12 19c-3.87 0-7-3.13-7-7 0-3.53 2.61-6.43 6-6.92V2.05c-5.06.5-9 4.76-9 9.95 0 5.52 4.47 10 9.99 10 3.31 0 6.24-1.61 8.06-4.09l-2.6-1.53C16.17 17.98 14.21 19 12 19z"
                />
              </svg>
            </span>
            Project Report
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-req3s6-MuiTypography-root-MuiLink-root"
            href="/login"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
            </span>
            Log out
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Navigation tests is a basic snapshot with a width of less than 900 pixels 1`] = `
<div
  class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-1tvu1zq-MuiModal-root-MuiDrawer-root"
  data-testid="navigation"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft css-4t3x6l-MuiPaper-root-MuiDrawer-paper"
    tabindex="-1"
  >
    <div
      class="MuiBox-root css-1ndpvz5"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        close navigation
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <nav
      class="MuiBox-root css-10klw3m"
    >
      <ul
        class="MuiList-root MuiList-padding css-t39185-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            aria-current="page"
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root active"
            href="/contacts"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ContactsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 0H4v2h16V0zM4 24h16v-2H4v2zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-8 2.75c1.24 0 2.25 1.01 2.25 2.25s-1.01 2.25-2.25 2.25S9.75 10.24 9.75 9 10.76 6.75 12 6.75zM17 17H7v-1.5c0-1.67 3.33-2.5 5-2.5s5 .83 5 2.5V17z"
                />
              </svg>
            </span>
            Total Contacts
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
            href="/calendar"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DateRangeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                />
              </svg>
            </span>
            Calendar
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
            href="/project-report"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DataUsageIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 2.05v3.03c3.39.49 6 3.39 6 6.92 0 .9-.18 1.75-.48 2.54l2.6 1.53c.56-1.24.88-2.62.88-4.07 0-5.18-3.95-9.45-9-9.95zM12 19c-3.87 0-7-3.13-7-7 0-3.53 2.61-6.43 6-6.92V2.05c-5.06.5-9 4.76-9 9.95 0 5.52 4.47 10 9.99 10 3.31 0 6.24-1.61 8.06-4.09l-2.6-1.53C16.17 17.98 14.21 19 12 19z"
                />
              </svg>
            </span>
            Project Report
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-req3s6-MuiTypography-root-MuiLink-root"
            href="/login"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
            </span>
            Log out
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Navigation tests is a basic snapshot with a width of more than 900 pixels 1`] = `
<div
  class="MuiDrawer-root MuiDrawer-docked css-127lbmo-MuiDrawer-docked"
  data-testid="navigation"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
  >
    <div
      class="MuiBox-root css-1ndpvz5"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-h74647-MuiTypography-root-MuiLink-root"
        href="/contacts"
      >
        <img
          alt="logo"
          class="css-12pywbb"
          src="assets"
        />
      </a>
    </div>
    <nav
      class="MuiBox-root css-10klw3m"
    >
      <ul
        class="MuiList-root MuiList-padding css-t39185-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            aria-current="page"
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root active"
            href="/contacts"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ContactsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 0H4v2h16V0zM4 24h16v-2H4v2zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-8 2.75c1.24 0 2.25 1.01 2.25 2.25s-1.01 2.25-2.25 2.25S9.75 10.24 9.75 9 10.76 6.75 12 6.75zM17 17H7v-1.5c0-1.67 3.33-2.5 5-2.5s5 .83 5 2.5V17z"
                />
              </svg>
            </span>
            Total Contacts
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
            href="/calendar"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DateRangeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                />
              </svg>
            </span>
            Calendar
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eb8jze-MuiTypography-root-MuiLink-root"
            href="/project-report"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DataUsageIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 2.05v3.03c3.39.49 6 3.39 6 6.92 0 .9-.18 1.75-.48 2.54l2.6 1.53c.56-1.24.88-2.62.88-4.07 0-5.18-3.95-9.45-9-9.95zM12 19c-3.87 0-7-3.13-7-7 0-3.53 2.61-6.43 6-6.92V2.05c-5.06.5-9 4.76-9 9.95 0 5.52 4.47 10 9.99 10 3.31 0 6.24-1.61 8.06-4.09l-2.6-1.53C16.17 17.98 14.21 19 12 19z"
                />
              </svg>
            </span>
            Project Report
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-b7k2hd-MuiListItem-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-req3s6-MuiTypography-root-MuiLink-root"
            href="/login"
          >
            <span
              class="MuiBox-root css-tyy0l3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
            </span>
            Log out
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Navigation tests is a snapshot with the open prop equal to false 1`] = `null`;
